services/auth/middleware.go

```diff
@@ -58,3 +61,182 @@ func authShared(ctx *context.Context, authMethod Method) error {
 	}
 	return nil
 }
+
+// ToggleOptions contains required or check options
+type ToggleOptions struct {
+	SignInRequired  bool
+	SignOutRequired bool
+	AdminRequired   bool
+	DisableCSRF     bool
+}
+
+// Toggle returns toggle options as middleware
+func Toggle(options *ToggleOptions) func(ctx *context.Context) {
```

Comment:
A nit, could we have a new name for `Toggle` since here is a refactoring chance?

Everything I see some requests blocked by `Toggle` I would spent some time on thinking about what happens ...

Reply:
Thought about that too. Do you have a good name for the method? `VerifyAuthOptions` or something like that?

Reply:
Maybe `VerifyAuthWithOptions` ?

Either is better than `Toggle`.